### ISSUE TITLE
Typescript error upgrading from 2.15.6 to 2.16.1

### ISSUE THREAD

**Issue by howieweiner** [aws-amplify/amplify-ui#1845]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Authenticator

### How is your app built?

CRA

### Please describe your bug.

Hi. I've just bumped the minor version from 2.15.6 to 2.16.1 and now the Authenticator component throws a Typescript typecheck error as follows:

TS2339: Property 'user' does not exist on type '{ signOut?: ((data?: AuthEventData | undefined) => void) | undefined; user?: CognitoUserAmplify | undefined; } | undefined'.

This is being cuased when attempting to access the user property as described in the docs:

```
<Authenticator>
        {({ user }) => {
          return (
...
```

Confusingly, the error message implies that user does exist on the type

Looks like this was introduced with a refactor in 2.16.0
https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CHANGELOG.md#2160

I've downgraded to 2.15.8 and there is no issue

### What's the expected behaviour?

No TS errors

### Help us reproduce the bug!

Implement a simple app following the docs.. but using TypeScript
https://ui.docs.amplify.aws/components/authenticator?platform=react

### Code Snippet

```typescript
// Put your code below this line.

```

**Comment by calebpollman** [comment]
Thanks for opening this @howieweiner. Was able to replicate your issue locally and will get out a fix

**Comment by SunnieBB** [comment]
Hi there, I'm having the same issue, and because of this my builds are unsuccessful.

```
Property 'signOut' does not exist on type '{ signOut?: ((data?: AuthEventData | undefined) => void) | undefined; user?: CognitoUserAmplify | undefined; } | undefined'.

160 |           >
                                 161 |             {
                                 > 162 |               ({ signOut, user }) => {
                                 |                  ^
                                 163 |
                                 164 |                 return (
                                 165 |     
```

**Comment by wlee221** [comment]
Hi @SunnieBB and @howieweiner, we've merged the fix #1850 and will be releasing very soon today. I'll keep you updated!

**Comment by wlee221** [comment]
Fix now has been released to `@aws-amplift/ui-react@2.16.2`. Please fee free to reopen if issue persists 🙏

**Comment by howieweiner** [comment]
Hi @wlee221 @calebpollman Thanks for the qjick turn around. Can confirm that the lates release fixes the issue for me
